pubsub/pubsub.go

```diff
@@ -534,3 +536,30 @@ func wrapError(ec errorCoder, err error) error {
 	}
 	return gcerr.New(ec.ErrorCode(err), err, 2, "pubsub")
 }
+
+// RunWorkerPool repeatedly fetches messages from the given subscription sub, in an
```

Comment:
This is a nice thing, but I'm not sure it should be in the `pubsub` package. It's really a utility function or helper, written on top of `pubsub`. I can imagine other helpers, like one that automatically extends the ack deadline (when we implement that ability). I'm thinking that to keep the pubsub API small and simple, we should put those kinds of things in a separate package.

If we *do* keep this in this package, would it make more sense as a method of `Subscription`?

Reply:
Thinking some more about this, I feel even more strongly that it should be in a different package. There are reasons we went with the `Subscription.Receive` function we have now: it's easy to understand, has no magic, and is symmetric with `Topic.Send`. This is definitely a "use at your own risk" kind of function. You have to be careful that your callback is goroutine-safe, for instance, and it can't pass the message off to another goroutine for processing and acking unless it blocks until the ack.

Reply:
Sounds reasonable. Giving that a shot now.

Reply:
Do you have any suggestion as far as the naming for this new package?

How about
```
import "gocloud.dev/pubsub/subs"

subs.ReceiveConcurrently(...)
```
?

Reply:
I'd go with `pubsub/pshelpers` or `pubsub/psutil` since I see this as a place to put other conveniences, like the aforementioned ack deadline extender. I think of the new package as something like `io/ioutil` or `github.com/google/cmp/cmpopts`.

BTW, it's a good idea to include at least a little bit of the parent package in the name (IOutil, CMPopts, etc.) since you end up using only the last component of the path in the importing file, and if everyone used `opts` or `util` or `testing`, it would be confusing.

Reply:
Okay, `pubsub/psutil` sounds fine.